src/transform/CollapseTable.js

```diff
@@ -4,33 +4,58 @@ import elementUtilities from './ElementUtilities'
 
 const SECTION_TOGGLED_EVENT_TYPE = 'section-toggled'
 
+/**
+ * Determine if we want to extract text from this header.
+ * @param {!Element} header
+ * @return {!boolean}
+ */
+const isHeaderEligible = header => Polyfill.querySelectorAll(header, 'a').length < 3
+
+/**
+ * Extracts header text.
+ * @param {!Document} document
+ * @param {!Element} header
+ * @return {?string}
+ */
+const extractHeaderText = (document, header) => {
+  // Clone header into fragment. This is done so we can remove some elements we don't want
+  // represented when "textContent" is used. Because we've cloned the header into a fragment, we are
+  // free to strip out anything we want without worrying about affecting the visible document.
+  const fragment = document.createDocumentFragment()
+  fragment.appendChild(header.cloneNode(true)) // eslint-disable-line require-jsdoc
+  Polyfill.querySelectorAll(fragment, '.geo, .coordinates, sup.reference')
+    .forEach(el => el.remove())
+  return fragment.textContent.trim()
+}
+
 /**
  * Find an array of table header (TH) contents. If there are no TH elements in
  * the table or the header's link matches pageTitle, an empty array is returned.
+ * @param {!Document} document
  * @param {!Element} element
  * @param {?string} pageTitle Unencoded page title; if this title matches the
  *                            contents of the header exactly, it will be omitted.
  * @return {!Array<string>}
  */
-const getTableHeader = (element, pageTitle) => {
-  const thArray = []
+const getTableHeaderTextArray = (document, element, pageTitle) => {
+  const headerTextArray = []
   const headers = Polyfill.querySelectorAll(element, 'th')
   for (let i = 0; i < headers.length; ++i) {
     const header = headers[i]
-    const anchors = Polyfill.querySelectorAll(header, 'a')
-    if (anchors.length < 3) {
+    if (isHeaderEligible(header)) {
+      const headerText = extractHeaderText(document, header)
       // Also ignore it if it's identical to the page title.
-      if ((header.textContent && header.textContent.length) > 0
-        && header.textContent !== pageTitle && header.innerHTML !== pageTitle) {
-        thArray.push(header.textContent)
+      if ((headerText && headerText.length) > 0
```

Comment:
I know this was here before but since you're changing the line, consider replacing `(headerText && headerText.length) > 0` with just `headerText`.

Reply:
Wouldn't that not catch zero length strings?

Reply:
`Boolean('')` is falsy so it should be fine.

Reply:
I updated code related to this and added more tests :)